File: app/api/auth/auth_api_routes.py
from flask import Blueprint, jsonify, request
from app.services.auth_services_routes import authenticate_user_with_cognito, verify_mfa_code, create_access_token

auth_api_routes = Blueprint('auth_api_routes', __name__)

@auth_api_routes.route("/login", methods=["POST"])
def login():
    data = request.json
    username = data.get('username')
    password = data.get('password')

    if not username or not password:
        return jsonify({"detail": "Username and password are required"}), 400

    auth_response = authenticate_user_with_cognito(username, password)
    
    if auth_response.get("mfa_required"):
        return jsonify({"mfa_required": True, "session": auth_response["session"]})

    authentication_result = auth_response.get("authentication_result")
    if not authentication_result:
        return jsonify({"detail": "Invalid credentials"}), 401

    access_token = create_access_token({
        "sub": username,
        "role": "employee",
        "email": username
    })

    return jsonify({"token": access_token, "role": "employee", "email": username, "mfa_required": False})

@auth_api_routes.route("/verify-mfa", methods=["POST"])
def verify_mfa_endpoint():
    data = request.json
    session = data.get('session')
    code = data.get('code')

    if not (session and code):
        return jsonify({"detail": "Session and code are required"}), 400

    try:
        auth_result = verify_mfa_code(session, code)
        access_token = create_access_token({
            "sub": "username-placeholder",
            "role": "employee"
        })
        return jsonify({"token": access_token, "role": "employee", "email": "username-placeholder", "mfa_required": False})
    except Exception as e:
        return jsonify({"detail": "MFA verification failed"}), 401

File: app/api/auth/auth_api_routes.js
// app/auth/auth_api_routes.js

export default async function handler(req, res) {
  // Allow only POST requests
  if (req.method !== "POST") {
    return res.status(405).json({ detail: "Method Not Allowed" });
  }

  // Destructure the request body
  const { username, password } = req.body;

  // Validate request data
  if (!username || !password) {
    return res.status(400).json({ detail: "Username and password are required" });
  }

  // API endpoint fetched from environment variable 
  const apiBaseUrl = process.env.NEXT_PUBLIC_API_URL;
  
  // Use the correct endpoint path based on backend route configuration
  // Changed from /api/auth/authenticate to /api/user/login to match the backend routes
  const loginEndpoint = `${apiBaseUrl}/api/user/login`;
  
  console.log(`Attempting to connect to API at: ${loginEndpoint}`);

  try {
    // Make a POST request to the Flask backend with additional options
    const response = await fetch(loginEndpoint, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "Accept": "application/json",
        "Origin": req.headers.origin || "https://console-encryptgate.net"
      },
      body: JSON.stringify({ username, password }),
      credentials: "include",
      mode: "cors",
      timeout: 10000 // 10 second timeout
    });

    // Handle non-OK responses
    if (!response.ok) {
      const errorData = await response.json().catch(() => ({ 
        detail: `HTTP error! Status: ${response.status}`,
        status: response.status
      }));
      console.error("Authentication API Error:", errorData);
      return res.status(response.status).json(errorData);
    }

    // Forward successful response to the client
    const data = await response.json();
    res.status(200).json(data);

  } catch (error) {
    // Log and return a server error response with more detailed information
    console.error("Login API Error:", error.message || error);
    res.status(500).json({ 
      detail: "Failed to connect to authentication service. Please try again later.",
      error: error.message || "Unknown error",
      endpoint: loginEndpoint
    });
  }
}